javascript-calculator/src/reducers/calculateReducer.js

```diff
@@ -0,0 +1,115 @@
+import { NUMBER_INPUT, OPERATOR_INPUT, DECIMAL_INPUT, CALCULATE, CLEAR } from '../actions';
+//import { Caption } from 'react-bootstrap/lib/Carousel';
+
+const initState = {
+    display: "0",
+    equation: "",
+};
+const calculateReducer = (prevState = initState, action) => {
+    console.log(prevState)
+    let { display, equation } = prevState
+    switch (action.type) {
+        case NUMBER_INPUT:
+            if (prevState.display === "0" || prevState.equation.includes("=")) {
+                display = action.payload
+                equation = action.payload
+            } else if (prevState.equation.match(/[+\-*/]$/)) {
+                const value = prevState.equation + action.payload;
+                display = action.payload
+                equation = value
+            } else if (prevState.equation.match(/[0-9.]$/)) {
+                if (prevState.equation.match(/[+\-*/]/) == null && !prevState.equation.includes("=")) {
+                    const value = prevState.equation + action.payload;
+                    display = value
+                    equation = value
+                } else {
+                    display = prevState.display + action.payload
+                    equation = prevState.equation + action.payload
+                }
+            };
+            return { display, equation }
```

Comment:
**We don't mutate the state.** You can create a copy with Object.assign() by doing `return Object.assign( {} ,prevState, { display, equation })` **Note:** You must supply an empty object as the first parameter or it will mutate the first argument. You can also enable the object spread operator proposal to write `return { ...prevState, display, equation }` instead. More information on: [https://redux.js.org/basics/reducers](url)